src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8302,7 +8300,162 @@ private ErrorPropertySymbol CreateErrorPropertySymbol(ImmutableArray<PropertySym
             }
         }
 
-        internal bool ReportDelegateInvokeUseSiteDiagnostic(BindingDiagnosticBag diagnostics, TypeSymbol possibleDelegateType,
+#nullable enable
+        internal NamedTypeSymbol? GetMethodGroupDelegateType(BoundMethodGroup node, BindingDiagnosticBag diagnostics)
+        {
+            if (GetUniqueSignatureFromMethodGroup(node, diagnostics) is { } method &&
+                GetMethodGroupOrLambdaDelegateType(method.RefKind, method.ReturnsVoid ? default : method.ReturnTypeWithAnnotations, method.ParameterRefKinds, method.ParameterTypesWithAnnotations, diagnostics) is { } delegateType)
+            {
+                return delegateType;
+            }
+            return null;
+        }
+
+        private static readonly MemberSignatureComparer s_uniqueSignatureFromMethodGroupComparer = new MemberSignatureComparer(
```

Comment:
nit: consider moving to `MemberSignatureComparer` (along with other instances) and making the constructor private again